### PR TITLE
CR-1076568 fixes

### PR DESCRIPTION
**Pull request by sarab96**
CR fixes

### CODE REVIEW

File: src/xma/src/xmaplugin/xmaplugin.cpp [hunk 1]
```diff
@@ -994,7 +985,7 @@ int32_t xma_plg_cu_cmd_status(XmaSession s_handle, XmaCUCmdObj* cmd_obj_array, i
         } else if (!all_done) {
             if (g_xma_singleton->cpu_mode == XMA_CPU_MODE1) {
                 std::unique_lock<std::mutex> lk(priv1->m_mutex);
-                priv1->work_item_done_1plus.wait(lk);
+                priv1->kernel_done_or_free.wait(lk);
                 lk.unlock();
```
Comment (stsoe) [inline src/xma/src/xmaplugin/xmaplugin.cpp:989]: remove

File: src/xma/src/xmaplugin/xmaplugin.cpp [hunk 2]
```diff
@@ -994,7 +985,7 @@ int32_t xma_plg_cu_cmd_status(XmaSession s_handle, XmaCUCmdObj* cmd_obj_array, i
         } else if (!all_done) {
             if (g_xma_singleton->cpu_mode == XMA_CPU_MODE1) {
                 std::unique_lock<std::mutex> lk(priv1->m_mutex);
-                priv1->work_item_done_1plus.wait(lk);
+                priv1->kernel_done_or_free.wait(lk);
```
Comment (stsoe) [inline src/xma/src/xmaplugin/xmaplugin.cpp:988]: What happens if this condition wakes up spuriously?

**Review by stsoe — COMMENTED** [review]
Minor comments.

File: src/xma/src/xmaplugin/xmaplugin.cpp (hunk 1 above)
Comment (sarab96) [reply src/xma/src/xmaplugin/xmaplugin.cpp]: done

File: src/xma/src/xmaplugin/xmaplugin.cpp (hunk 2 above)
Comment (sarab96) [reply src/xma/src/xmaplugin/xmaplugin.cpp:988]: In that case it will redo all check and go back to wait if required

**Review by stsoe — APPROVED** [review]

**Pull request merged**